pkg/awsutils/awsutils.go

```diff
@@ -883,6 +886,34 @@ func (cache *EC2InstanceMetadataCache) AllocIPAddresses(eniID string, numIPs int
 	return nil
 }
 
+// DeallocIPAddresses allocates numIPs of IP address on an ENI
+func (cache *EC2InstanceMetadataCache) DeallocIPAddresses(eniID string, ips []string) error {
+	ctx := context.Background()
```

Comment:
Currently just using a dummy context, but in a future change we should pipe context from higher up so that we can take advantage of cancellation.